docs/howtos/nodeenv.md

```diff
@@ -0,0 +1,38 @@
+## Instructions
+Once you've created a python virtual environment, you can use `nodeenv` to install particular versions of node.js within the environment. This allows you to use a different node.js version in the virtual environment than what's available on your host, keep multiple virtual enviroments with different versions of node.js, and to install node.js "global" modules that are only available within the virtual environment.
+
+First make sure your virtual environment is activated. With pyenv you can do this with:
+```bash
+$ pyenv activate kolibri-py3.9
+```
+
+If nodeenv is not already installed in your virtual environment, you can install it using this command:
+```bash
+$ pip install nodeenv
+```
+
+If you don't already know what version you need to install, the first step is to determine the latest node.js version for a major release version. You can use `nodeenv` to list out all versions:
+```bash
+$ nodeenv -l
+```
+but this lists out everything. Alternatively, here's a one line bash function that can be used to determine that version:
+```bash
+$ function latest-node() { curl -s "https://nodejs.org/dist/latest-v$1.x/" | egrep -m 1 -o "$1\.[0-9]+\.[0-9]+" | head -1; }
+$ latest-node 16
+16.16.0
+```
+
+Once you've determined the version, you can install it:
+```bash
+$ nodeenv --python-virtualenv --node 16.16.0
+ * Install prebuilt node (16.16.0) ..... done.
+ * Appending data to /home/bjester/Projects/learningequality/kolibri/venv/bin/activate
+ * Appending data to /home/bjester/Projects/learningequality/kolibri/venv/bin/activate.fish
+```
+
+You'll notice in the output above, the installation modifies the virtual environment activation scripts. Reloading the virtual environment will ensure everything works correctly.
+```bash
+$ source deactivate
```

Comment:
This should be ``pyenv deactivate``, right?

Reply:
Both will work - the slight advantage of `source deactivate` is that it works in multiple Virtual environment managers.

Reply:
``source deactivate`` was not successful in closing the virtual environment:

```
kolibrien@kolibriU20:~$ cd LE/kolibri/
kolibrien@kolibriU20:~/LE/kolibri$ pyenv activate kolibri-py3.9
pyenv-virtualenv: prompt changing will be removed from future release. configure `export PYENV_VIRTUALENV_DISABLE_PROMPT=1' to simulate the behavior.
(kolibri-py3.9) kolibrien@kolibriU20:~/LE/kolibri$ source deactivate
bash: deactivate: No such file or directory
(kolibri-py3.9) kolibrien@kolibriU20:~/LE/kolibri$ pyenv deactivate
kolibrien@kolibriU20:~/LE/kolibri$ 
```

Reply:
Updated to the pyenv specific command as suggested.